pkg/controllers/dynakube/istio/reconciler.go

```diff
@@ -38,6 +40,58 @@ func NewReconciler(istio *Client) Reconciler {
 	}
 }
 
+func (r *reconciler) ReconcileCSIDriver(ctx context.Context, dynakube *dynatracev1beta2.DynaKube) error {
+	log.Info("reconciling istio components for the CSI driver")
+
+	if dynakube == nil {
+		return errors.New("can't reconcile csi driver of nil dynakube")
+	}
+
+	codeModulesURL := dynakube.CodeModulesImage()
+
+	parsedCodeModulesURL, err := parseCodeModulesImageURL(codeModulesURL)
+	if err != nil {
+		return err
+	}
+
+	codeModulesHost, err := dtclient.ParseEndpoint(parsedCodeModulesURL)
+	if err != nil {
+		return err
+	}
+
+	err = r.reconcileCommunicationHosts(ctx, []dtclient.CommunicationHost{codeModulesHost}, CSIDiverComponent)
+	if err != nil {
+		return errors.WithMessage(err, "error reconciling config for codeModulesImage")
+	}
+
+	log.Info("reconciled istio objects for CSI driver")
+
+	return nil
+}
+
+func parseCodeModulesImageURL(rawUrl string) (string, error) {
+	parsedURL, err := url.Parse(rawUrl)
+	if err != nil {
+		return "", errors.New("can't parse the codeModules image URL")
+	}
+
+	if parsedURL.Scheme == "" {
+		parsedURL.Scheme = "https"
+
+		if !strings.HasPrefix(rawUrl, "//") {
+			// if no scheme at all is set we want to add this prefix
+			rawUrl = "//" + rawUrl
+		}
```

Comment:
maybe I am not getting something here, but why is this necessary?

pro-tip: `parseCodeModulesImageURL` is nice and easily testable function, writing a bunch of unittests with all the weird possible `rawUrl` examples might help understanding (for all parties) 😉

Reply:
No you are right, this rules out my logic tbh 😅  will remove it and add proper tests 👍

Reply:
Note: It is indeed necessary now I remember again why: if I have some `url` that looks like this: `some.test.url `and do not have this if check the function it will concatenate the scheme which will be `https` in this case with `:` and the `rawUrl`, but I will fix it by simply concatenating the scheme with the string `://` and the `rawUrl`, this does the same.

Reply:
pls have a look at the tests and especially the case where i test `//some.url.com/test` 

Tell me if this is a actually possible case. Because if not I can remove the logic which you questioned again, because for now I let it in.

https://github.com/Dynatrace/dynatrace-operator/pull/3343/commits/8075a6597276e226add71a643131b960503aa5e8

Reply:
I have no idea when `//some.url.com/test` would happen and what it means

Reply:
will remove it

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3343/commits/619e50a181dc45bb873290e4cd0f59372eadecdc